src/main/java/org/example/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+package org.example.bookstore.service.order;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.example.bookstore.dto.orderdto.OrderRequestDto;
+import org.example.bookstore.dto.orderdto.OrderResponseDto;
+import org.example.bookstore.dto.orderdto.OrderStatusUpdateRequestDto;
+import org.example.bookstore.dto.orderdto.OrderUpdatedDto;
+import org.example.bookstore.dto.orderitemdto.OrderItemResponseDto;
+import org.example.bookstore.exception.EmptyCartException;
+import org.example.bookstore.exception.EntityNotFoundException;
+import org.example.bookstore.mapper.OrderItemsMapper;
+import org.example.bookstore.mapper.OrderMapper;
+import org.example.bookstore.model.CartItem;
+import org.example.bookstore.model.OrderItem;
+import org.example.bookstore.model.ShoppingCart;
+import org.example.bookstore.model.User;
+import org.example.bookstore.model.order.Order;
+import org.example.bookstore.model.status.Status;
+import org.example.bookstore.repository.cartitem.CartItemRepository;
+import org.example.bookstore.repository.order.OrderRepository;
+import org.example.bookstore.repository.orderitem.OrderItemRepository;
+import org.example.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import org.example.bookstore.repository.user.UserRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemsMapper orderItemsMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public OrderResponseDto createNewOrder(OrderRequestDto requestDto,
+                                           Long userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("User with id: " + userId
+                        + "does not exist"));
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findShoppingCartByUserId(user.getId()).orElseThrow(()
+                        -> new EntityNotFoundException("Shopping cart does not exist"));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        if (cartItems == null || cartItems.isEmpty()) {
+            throw new EmptyCartException("Your shopping cart is empty ");
+        }
+
+        Order order = new Order();
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
```

Comment:
You can use `@CreationTimestamp` in the entity class instead of setting it manually here, or a `@PrePersist` method there